src/Microsoft.AspNetCore.Authentication.Abstractions/AuthenticationProperties.cs

```diff
@@ -109,6 +127,40 @@ protected void SetString(string key, string value)
             }
         }
 
+        /// <summary>
+        /// Get a parameter from the <see cref="Parameters"/> collection.
+        /// </summary>
+        /// <typeparam name="T">Parameter type.</typeparam>
+        /// <param name="key">Parameter key.</param>
+        /// <returns>Retrieved value or the default value if the property is not set.</returns>
+        public T GetParameter<T>(string key)
+        {
+            return Parameters.TryGetValue(key, out var obj) && obj is T value ? value : default;
+        }
+
+        /// <summary>
+        /// Set a parameter value in the <see cref="Parameters"/> collection.
+        /// </summary>
+        /// <typeparam name="T">Parameter type,</typeparam>
+        /// <param name="key">Parameter key.</param>
+        /// <param name="value">Value to set or <c>null</c> to remove the property.</param>
+        public void SetParameter<T>(string key, T value)
+        {
+            if (value != null)
+            {
+                Parameters[key] = value;
+            }
+            else if (Parameters.ContainsKey(key))
```

Comment:
Do we really need this null clearing logic?  Can't they just do Parameters.Remove(key) directly instead of set with null?

Reply:
That was mostly to get a symmetric behavior to the Set~ methods on the `Items` dictionary.

Reply:
I don't think that's an important concern as the Items are used for serialization where this behavior is much more important

Reply:
That’s a good point! I’ll remove that part then.